DS13/code/modules/overmap/components.dm

```diff
@@ -0,0 +1,643 @@
+			return update_icon()
+		if(I.tool_behaviour == TOOL_SCREWDRIVER && repair_step == "screwdriver")
+			playsound(loc,I.usesound,100,1)
+			to_chat(user, "<span class='notice'>You start to mend [src]'s isolinear circuitry.</span>")
+			if(do_after(user, 30, target = src))
+				repair_step = "wirecutters"
+			return update_icon()
+		if(I.tool_behaviour == TOOL_WIRECUTTER && repair_step == "wirecutters")
+			playsound(loc,I.usesound,100,1)
+			to_chat(user, "<span class='notice'>You start cutting out [src]'s fused relays.</span>")
+			if(do_after(user, 30, target = src))
+				repair_step = "screwdriver2"
+				to_chat(user, "<span class='notice'>You have successfully repaired [src]! re-assembly is the reverse of removal.</span>")
+				obj_integrity = max_integrity
+			return update_icon()
+		if(I.tool_behaviour == TOOL_SCREWDRIVER && repair_step == "screwdriver2")
+			playsound(loc,I.usesound,100,1)
+			to_chat(user, "<span class='notice'>You start fasten [src]'s screws.</span>")
+			if(do_after(user, 30, target = src))
+				repair_step = "wrench2"
+			return update_icon()
+		if(I.tool_behaviour == TOOL_WRENCH && repair_step == "wrench2")
+			playsound(loc,I.usesound,100,1)
+			to_chat(user, "<span class='notice'>You start to re-fasten [src]'s couplings.</span>")
+			if(do_after(user, 30, target = src))
+				repair_step = "wrench"
+				to_chat(user, "<span class='notice'>You have successfully re-assembled [src]! replace its cover to complete repairs.</span>")
+			return update_icon()
+	. = ..()
+	update_icon()
+
+/obj/structure/overmap_component/plasma_relay/examine(mob/user)
+	. = ..()
+	if(supplier && supplying)
+		to_chat(user, "it receiving plasma from [supplier] and is supplying the <b>[supplying]</b> system for [linked]")
+	else
+		to_chat(user, "-it's currently <b>unpowered</b>")
+
+/obj/structure/overmap_component/plasma_relay/take_damage(damage_amount)
+	. = ..()
+	var/sound = pick(zaps)
+	playsound(src.loc, sound, 70,1)
+	var/bleep = pick(bleeps)
+	playsound(src.loc, bleep, 70,1)
+	do_sparks(5, 8, src)
+	if(plasma_volume >= 2)
+		var/turf/T = get_step(src, SOUTH)
+		T.atmos_spawn_air("plasma=10;TEMP=2000") //Anything less than this barely even matters
+		plasma_volume -= 2
+
+/obj/structure/overmap_component/plasma_relay/update_icon()
+	if(panel_open)
+		icon_state = "relay-open"
+		return
+	else
+		icon_state = "relay-closed"
+		if(obj_integrity < 40)
+			icon_state = "relay-closed-alert"
+			set_active(FALSE)
+		return
+	icon_state = "relay-closed"
+
+/obj/structure/overmap_component/plasma_relay/process()
+	update_icon()
+	if(obj_integrity < 40 || plasma_volume < plasma_drain)
+		set_active(FALSE)
+		return
+	if(!supplying)
+		find_supplying()
+		return
+	if(!active && !panel_open)
+		set_active(TRUE)
+		return
+	if(plasma_volume >= plasma_drain)
+		plasma_volume -= plasma_drain
+		set_active(TRUE)
+	else
+		set_active(FALSE)
+
+/obj/structure/overmap_component/plasma_relay/proc/set_active(var/bool, var/forced = FALSE)
```

Comment:
bool is a fucking terrible varname, and you know it.

Reply:
I forgot what it does 😎

Reply:
exactly.